From: Matthias Brugger <mbrugger@suse.com>
Date: Thu, 3 Nov 2016 20:23:11 +0000
Subject: arm64: fix includes for smp.h
References: fate#322150
Patch-mainline: Never, fixes local compile issue

Mainline commit 
57c82954e77f ("arm64: make cpu number a percpu variable")
includes percpu.h into smp.h which is needed to access functions and macros in
subsequent patches. Include percpu.h only to fix the following compile errors:

In file included from ./arch/arm64/include/asm/timex.h:19:0,
                 from include/linux/timex.h:65,
                 from include/linux/sched.h:19,
                 from arch/arm64/kernel/asm-offsets.c:21:
./arch/arm64/include/asm/arch_timer.h:62:3: error: unknown type name 'timer_unstable_counter_workaround'
   timer_unstable_counter_workaround);
   ^
./arch/arm64/include/asm/arch_timer.h: In function 'arch_timer_reg_read_cp15':
./arch/arm64/include/asm/arch_timer.h:70:8: error: implicit declaration of function '__this_cpu_read' [-Werror=implicit-function-declaration]
   wa = __this_cpu_read(timer_unstable_counter_workaround); \
        ^
./arch/arm64/include/asm/arch_timer.h:121:11: note: in expansion of macro 'arch_timer_reg_read_stable'
    return arch_timer_reg_read_stable(cntp_tval_el0);
           ^
./arch/arm64/include/asm/arch_timer.h:70:24: error: 'timer_unstable_counter_workaround' undeclared (first use in this function)
   wa = __this_cpu_read(timer_unstable_counter_workaround); \
                        ^
./arch/arm64/include/asm/arch_timer.h:121:11: note: in expansion of macro 'arch_timer_reg_read_stable'
    return arch_timer_reg_read_stable(cntp_tval_el0);
           ^
./arch/arm64/include/asm/arch_timer.h:70:24: note: each undeclared identifier is reported only once for each function it appears in
   wa = __this_cpu_read(timer_unstable_counter_workaround); \
                        ^
./arch/arm64/include/asm/arch_timer.h:121:11: note: in expansion of macro 'arch_timer_reg_read_stable'
    return arch_timer_reg_read_stable(cntp_tval_el0);
           ^
./arch/arm64/include/asm/arch_timer.h: In function 'arch_counter_get_cntvct':
./arch/arm64/include/asm/arch_timer.h:70:24: error: 'timer_unstable_counter_workaround' undeclared (first use in this function)
   wa = __this_cpu_read(timer_unstable_counter_workaround); \
                        ^
./arch/arm64/include/asm/arch_timer.h:162:9: note: in expansion of macro 'arch_timer_reg_read_stable'
  return arch_timer_reg_read_stable(cntvct_el0);
         ^

Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 arch/arm64/include/asm/smp.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/arm64/include/asm/smp.h
+++ b/arch/arm64/include/asm/smp.h
@@ -29,6 +29,8 @@
 
 #ifndef __ASSEMBLY__
 
+#include <asm/percpu.h>
+
 #include <linux/threads.h>
 #include <linux/cpumask.h>
 #include <linux/thread_info.h>
